scripts/generate_cpp_dependency_json.py

```
@@ -0,0 +1,221 @@
+    dep = index.get(package_name, {})
+    repo = dep.get("git_repository", "").replace("${github_base_url}", "https://github.com")
+    return {"url": repo, "tag": dep.get("git_tag", "")}
+
+
+def check_oss_components(package_name: str) -> tuple[str, int] | None:
+    """Return (web_url, project_id) if package_name exists in oss-components, else None."""
+    project_path = urllib.parse.quote(f"{GITLAB_OSS_GROUP}/{package_name}", safe="")
+    url = f"{GITLAB_API_BASE}/projects/{project_path}"
+    req = urllib.request.Request(url, headers={"PRIVATE-TOKEN": TOKEN})
+    try:
+        with urllib.request.urlopen(req) as resp:
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)